build/Shared/HashCodeCombiner.cs

```diff
@@ -58,6 +58,15 @@ internal void AddObject<T>(T o)
             }
         }
 
+        // Optimization: for value types, we can avoid boxing "o" by skipping the null check
+        internal void AddStruct<T>(T o)
+            where T : struct
```

Comment:
@wrall 
It's your call whether to add it or not.

Reply:
That's a great idea for enforcement.  I'll make this change.  Note that it will pull in a bunch of other files that weren't showing up in my solution's restore, but could theoretically hit under some scenario.